engine-nextgen/src/main/kotlin/graphql/nadel/enginekt/transform/result/json/JsonNodePath.kt

```diff
@@ -22,7 +22,9 @@ data class JsonNodePath(
     }
 
     fun dropLast(n: Int): JsonNodePath {
-        return copy(segments = segments.dropLast(n))
+        return copy(
+            segments = segments.subList(0, segments.size - n),
```

Comment:
Using `subList` creates a wrapper around the original list that transforms the indices instead of cloning the list. This actually heavily speeds things up.